### PR TITLE
Deprecate `kedro test` and `kedro lint`

### LINKED ISSUE #1872: Deprecate `kedro lint` and `kedro test`
## Description
https://github.com/kedro-org/kedro/issues/1766

## Implementation
Deprecate `kedro lint` and `kedro test` in the same way the other development related CLI commands were deprecated in https://github.com/kedro-org/kedro/issues/1616

### PR DESCRIPTION
**Pull request by noklam**
Signed-off-by: Nok Chan <nok.lam.chan@quantumblack.com>

## Description

Close #1872 

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1873"></a>

### CODE REVIEW

File: kedro/framework/cli/project.py
Comment (deepyaman) [inline kedro/framework/cli/project.py:104]: Do we need to point out that users should call linters directly? Do we need to specify which ones?

File: kedro/framework/cli/project.py
Comment (deepyaman) [inline kedro/framework/cli/project.py:79]: Just to be clear, this means we are not wrapping `pytest` in a Kedro command, but we're still supporting `pytest`, right? Since, if we were just going to say "testing is up to you now," we shouldn't call out `pytest` in the message.

I think it's fine as-is, if I've understood correctly.

File: kedro/framework/cli/project.py
Comment (deepyaman) [inline kedro/framework/cli/project.py:76]: Nit: I'm overly picky, but it would be nice to be consistent on space between docstring and `deprecation_message = ...` and between `click.secho` and the remaining logic below, IMO. :) I checked the previously-deprecated stuff, and it's not consistent, so I don't have a recommendation one way or another--just generally prefer if it's all the same.

File: RELEASE.md
Comment (deepyaman) [inline RELEASE.md:19]:
```suggestion
* `kedro test` and `kedro lint` will be deprecated.
```

File: kedro/framework/cli/project.py
Comment (noklam) [reply kedro/framework/cli/project.py:79]: I think we will still have some tests in starters? This isn't too important though, as this deprecating message is for 0.18.x, we will still support `pytest` until 0.19.0, so this message is just asking people to use `pytest` instead of `kedro test`.
#1879 is a more detailed docs for how to use `pytest`.

File: kedro/framework/cli/project.py
Comment (noklam) [reply kedro/framework/cli/project.py:104]: I am less sure about this one, maybe it will become more clear when #1620 is done. I am fine we either wait until #1620 is done or just merge this one and add some links when the docs PRs are done.

File: kedro/framework/cli/project.py
Comment (noklam) [reply kedro/framework/cli/project.py:76]: I wish the linter would do this for me, but it didn't. I will make a change to the styling.

Comment (noklam) [reply kedro/framework/cli/project.py]: I move all the deprecated message at the top of the function, and a newline between the `click.secho` and the actual logic.

File: kedro/framework/cli/project.py
Comment (jmholzer) [reply kedro/framework/cli/project.py:78]: Determining whether or not tests will still be in starters is part of an open issue (#1849).

Comment (jmholzer) [reply kedro/framework/cli/project.py:78]: I personally wouldn't mention `pytest` as an alternative in this message, I think this also would keep things consistent with the linting message.

**Review by jmholzer — APPROVED** [review]
Happy to approve this PR as it is, though I left a comment on the deprecation of `kedro test`.

File: kedro/framework/cli/project.py
Comment (noklam) [reply kedro/framework/cli/project.py:78]: I am gonna leave it as is. I don't think it matters too much and I am fine with either option.

I checked in 0.17.x we did similar thing for `kedro install`
https://github.com/kedro-org/kedro/blob/80c0d3a47166c54f2bff4bed42cf63e06e65273c/kedro/framework/cli/project.py#L175-L180

**Pull request merged**
